srtcore/api.cpp

```diff
@@ -254,18 +254,24 @@ int CUDTUnited::startup()
 
    PacketFilter::globalInit();
 
-   //init CTimer::EventLock
-
    if (m_bGCStatus)
-      return true;
+      return 1;
 
    m_bClosing = false;
 
    setupMutex(m_GCStopLock, "GCStop");
    setupCond(m_GCStopCond, "GCStop");
    {
        ThreadName tn("SRT:GC");
-       pthread_create(&m_GCThread, NULL, garbageCollect, this);
+       try
+       {
+          CreateThread(m_GCThread, garbageCollect, this);
```

Comment:
I have doubts as to whether it is worth having exception handling for thread creation in our code, provided that they are OUR wrappers and we can shape them as we see fit. Exception handling makes little sense if you are obliged to wrap a single function call into try-catch statement. It might make sense here, if you can catch every case of thread creation on a higher level and turn it into an SRT exception so that it's handled higher as user error, or if you create a bunch of threads in the beginning and you want one common catch for all of them.